consoleme/lib/aws.py

```diff
@@ -1880,12 +1889,24 @@ def allowed_to_sync_role(
     """
     allowed_tags = config.get("roles.allowed_tags", {})
     allowed_arns = config.get("roles.allowed_arns", [])
-    if not allowed_tags and not allowed_arns:
+    allowed_tag_keys = config.get("roles.allowed_tag_keys", [])
+    if not allowed_tags and not allowed_arns and not allowed_tag_keys:
         return True
 
     if role_arn in allowed_arns:
         return True
 
+    # Convert list of role tag dicts to an array of tag keys
+    # ex:
+    # role_tags = [{'Key': 'consoleme-authorized', 'Value': 'consoleme_admins'},
+    # {'Key': 'Description', 'Value': 'ConsoleMe OSS Demo Role'}]
+    # so: actual_tag_keys = ['consoleme-authorized', 'Description']
+    actual_tag_keys = [d['Key'] for d in role_tags]
+
+    # If any allowed tag key exists in the role's actual_tags this condition will pass
+    if allowed_tag_keys and any(x in allowed_tag_keys for x in actual_tag_keys):
```

Comment:
We can eliminate the list comprehension above by changing this to:

```python
if allowed_tag_keys and any(x in role_tags for x in allowed_tag_keys):
```

That way, each iteration through `allowed_tag_keys` is a hash lookup in the `role_tags` dict.

Reply:
@patricksanders - Thanks for the quick feedback.

I've made a change to eliminate the list comprehension and combine into one I had to do it slightly different from above in order to get it to work though.

```python
if allowed_tag_keys and any(
    x for x in role_tags for y in allowed_tag_keys if x["Key"] == y
):
```

Reply:
🤦 I was thinking the tags were `{'key': 'value'}` dicts.

Apologies for the back-and-forth, but can you revert back to what you originally had? That's a more straightforward approach given the structure of the tag data.

Reply:
No problem. That's it reverted back.

Reply:
Fixed the issue that caused the build pre-commit step to fail.